screen/wallets/addMultisigStep2.js

```diff
@@ -257,62 +183,65 @@ const WalletsAddMultisigStep2 = () => {
       return setIsLoading(false);
     }
 
-    setIsProvideMnemonicsModalVisible(false);
     setIsLoading(false);
-    setImportText('');
-    setAskPassphrase(false);
 
     const cosignersCopy = [...cosigners];
     cosignersCopy.push([xpub, fp, path]);
     if (Platform.OS !== 'android') LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
     setCosigners(cosignersCopy);
   };
 
-  const useMnemonicPhrase = async () => {
-    setIsLoading(true);
+  const HashIt = function (s) {
+    return createHash('sha256').update(s).digest().toString('hex');
+  };
 
-    if (MultisigHDWallet.isXpubValid(importText)) {
-      return tryUsingXpub(importText);
+  const _onReadUniformResourceV2 = part => {
+    if (!decoder) decoder = new BlueURDecoder();
+    try {
+      decoder.receivePart(part);
+      if (decoder.isComplete()) {
+        const data = decoder.toString();
+        decoder = false; // nullify for future use (?)
+        onBarScanned({ data });
+      }
+    } catch (error) {}
+  };
```

Comment:
What is the advantage of having the decoder as a global variable?

Reply:
To be brutally honest, I don’t see any. For this one tried to stick to the Bluewallet’s implementation, but if it makes noise to you as well  I’ll be happy to look for another way

Reply:
Ok 😁  I probably would have created the decoder locally in the method.

Reply:
Sounds good to me. I can make the amend 👍

Reply:
Cool 👏

Reply:
so sorry I remember the reason of this. The decoder instance must live between multiple calls of `_onReadUniformResourceV2` for those strings that are encoded into multiple QR codes that change over time. Once all parts are processed by the decoder instance is when you can dispose it.

I'm not sure if it make sense, please let me know if the explanation is confusing

Reply:
Ok I see, makes sense, thanks for the explanation 👍